build warnings in ntlm.c when using GitHub Actions ubuntu-latest

Building libgit2 using GitHub Actions ubuntu-latest platform, these warnings appear:

```
Scanning dependencies of target ntlmclient
[ 97%] Building C object deps/ntlmclient/CMakeFiles/ntlmclient.dir/ntlm.c.o
/home/runner/work/x/x/build/git2go/v0.0.0-20200108224216-1a1276eb740d/libgit2/deps/ntlmclient/ntlm.c: In function ‘ntlm_client_negotiate’:
##[error]/home/runner/work/x/x/build/git2go/v0.0.0-20200108224216-1a1276eb740d/libgit2/deps/ntlmclient/ntlm.c:312:23: warning: ‘hostname_offset’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  out->buf[out->pos++] = (value & 0x000000ff);
  ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
##[error]/home/runner/work/x/x/build/git2go/v0.0.0-20200108224216-1a1276eb740d/libgit2/deps/ntlmclient/ntlm.c:590:23: note: ‘hostname_offset’ was declared here
  size_t hostname_len, hostname_offset, domain_len, domain_offset;
                       ^~~~~~~~~~~~~~~
##[error]/home/runner/work/x/x/build/git2go/v0.0.0-20200108224216-1a1276eb740d/libgit2/deps/ntlmclient/ntlm.c:348:3: warning: ‘domain_offset’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   write_int32(ntlm, out, (uint32_t)offset);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
##[error]/home/runner/work/x/x/build/git2go/v0.0.0-20200108224216-1a1276eb740d/libgit2/deps/ntlmclient/ntlm.c:590:52: note: ‘domain_offset’ was declared here
  size_t hostname_len, hostname_offset, domain_len, domain_offset;
                                                    ^~~~~~~~~~~~~
```

The warnings are spurious. Neither `domain_offset` nor `hostname_offset` is used uninitialized. Nevertheless, it'd be nice to silence the log noise.

The easiest workaround is probably zeroing those two variables early on in the function. I'm happy to send a PR to that effect if that would be helpful. (Should the PR go to here or to the upstream ntml repo?)

--- a/deps/ntlmclient/ntlm.c
+++ b/deps/ntlmclient/ntlm.c
@@ -587,7 +587,9 @@ int ntlm_client_negotiate(
 	size_t *out_len,
 	ntlm_client *ntlm)
 {
-	size_t hostname_len, hostname_offset, domain_len, domain_offset;
+	size_t hostname_len, domain_len;
+	size_t domain_offset = 0;
+	size_t hostname_offset = 0;
 	uint32_t flags = 0;
 
 	assert(out && out_len && ntlm);

